src/worker/browser-env/runner/test-runner/index.ts

```diff
@@ -59,15 +67,114 @@ export class TestRunner extends NodejsEnvTestRunner {
 
     async run(opts: WorkerTestRunnerRunOpts): Promise<WorkerRunTestResult> {
         this._runOpts = opts;
+        let error: Error | undefined = undefined;
 
-        const results = await super.run({ ...opts, ExecutionThreadCls: wrapExecutionThread(this._socket) });
+        try {
+            await super.prepareToRun(this._runOpts);
 
+            error = await this._runWithAbort<Error | undefined>(throwIfAborted => {
+                const ExecutionThreadCls = wrapExecutionThread(this._socket, throwIfAborted);
+
+                return super.runRunnables(ExecutionThreadCls) as Promise<Error | undefined>;
+            });
+        } catch (err) {
+            error = err as Error;
+        }
+
+        while (error instanceof AbortError) {
+            error = undefined;
```

Comment:
Lets not use explicit `undefined` as value. There are reasons: https://eslint.org/docs/latest/rules/no-undefined
We can substitute it with `null`